src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -31,13 +35,20 @@ const useLocalDatabaseHook = () => {
   const {channelInfo, channelList, channelMember, chat, user} = databaseListener;
 
   const initDb = async () => {
+    const isEnteringApp = Boolean(initialStartup?.id);
+
     const db = await LocalDatabase.getDBConnection();
-    await LocalDatabaseMigration.migrateDb();
+    if (isEnteringApp && migrationStatus === 'NOT_MIGRATED') {
+      await LocalDatabaseMigration.migrateDb();
+      setMigrationStatus('MIGRATED');
+    }
 
     setLocalDb(db);
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new logic introduces a conditional database migration based on the `initialStartup` and `migrationStatus` states. This is a good practice for managing database versions. However, ensure that error handling is implemented in case the migration fails.
```diff
+    const isEnteringApp = Boolean(initialStartup?.id);
+
+    const db = await LocalDatabase.getDBConnection();
+    if (isEnteringApp && migrationStatus === 'NOT_MIGRATED') {
+      await LocalDatabaseMigration.migrateDb();
+      setMigrationStatus('MIGRATED');
+    }
+  
+    setLocalDb(db);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->